features/0211-route-coordination/README.md

```diff
@@ -26,7 +30,9 @@ A recipient may discover an agent capable of routing using the Feature Discovery
 
 First, the _recipient_ sends a `route_request` message to the _router_. If the _router_ is willing to route messages, it will respond with a `route_grant` message. The _recipient_ will share the routing information in the grant message with other contacts.
 
-When a new key is used by the _recipient_, it must be registered with the _router_ to enable route identitifcation. This is done with a `keylist_update` message.
+If the _router_ requires the _recipient_  to agree to terms prior to service, a `route_deny` message will be returned listing the URIs of terms that the user must agree to. Term agreement is indicated by including the same URIs in the `terms` attribute of the `route_requeste` message. Terms the _recipient_ wish the _router_ to agree to are indicated in the `user_terms` attribute of the `route_request` message. The *router* may indicate which user terms they support in the `supported_user_terms` attribute of the `Route Deny` message.
```

Comment:
See my comment a few lines below about why I don't think we should be promoting the term "router" or "routing" here.

Reply:
Typo: "route_requeste" --> "route_request"

Reply:
On router vs mediator: This protocol could technically coordinate either mediators or relays. Relays are not _required_ to be known to the recipient, but they could be.

Reply:
Regards this - "Relays are not required to be known to the recipient, but they could be." If they are known to the recipient, they are by definition a mediator and not a relay. AFAIK - that is the difference between the two.

Reply:
Ooh, this is getting nuanced. :-)

A mediator *must* be known to both parties, because accommodations must be made in the crypto to account for mediation; these bidirectional accommodations are what makes them a mediator. A relay *could* be known, but generally wouldn't be, because there is no point. (Relays don't have keys and make no accommodations, so there's nothing to tell the other party about or agree to.) Thus, a party that's doing relay, that's known, still isn't a mediator if it doesn't need to be accounted for in the crypto.

To say it another way: the defining characteristic of a mediator is that they receive encrypted messages, *encrypted for them*, do a decryption, and then take intelligent action on the plaintext they see (typically the plaintext is a `forward` message with a further-encrypted payload). The mediator's behavior is something that the sender and recipient have both agreed upon.